python/python/benchmarks/test_search.py

```diff
@@ -57,6 +69,60 @@ def test_dataset(data_dir: Path) -> lance.LanceDataset:
     return dataset
 
 
+def create_delete_dataset(data_dir):
+    tmp_path = data_dir / "search_dataset_with_delete"
+    dataset = find_or_clean(tmp_path)
+    if dataset:
+        return dataset
+
+    clean_path = data_dir / "search_dataset"
+    shutil.copytree(clean_path, tmp_path)
+
+    dataset = lance.dataset(tmp_path)
+    dataset.delete("filterable % 2 != 0")
+
+    return dataset
+
+
+def create_new_rows_dataset(data_dir):
+    tmp_path = data_dir / "search_dataset_with_new_rows"
```

Comment:
In the future, we might consider just making each of these a different version of the same dataset. That way we don't have to manage all these directories.